inv/inventory.go

```diff
@@ -323,6 +324,28 @@ func (i *inventory) GetFiltersAttributes(ctx context.Context) ([]model.FilterAtt
 	return attributes, nil
 }
 
+func (i *inventory) DeleteGroup(
+	ctx context.Context,
+	groupName model.GroupName,
+) (*model.UpdateResult, error) {
+	deviceIDs, err := i.db.DeleteGroup(ctx, groupName)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to delete group")
+	}
+
+	res := &model.UpdateResult{}
+	for deviceID := range deviceIDs {
+		deviceIDStr := string(deviceID)
+		res.MatchedCount += 1
+		if deviceIDStr != "" {
+			res.UpdatedCount += 1
+			i.maybeTriggerReindex(ctx, deviceIDStr)
+		}
+	}
```

Comment:
Can we check if reporting exist outside instead of inside the loop?
```suggestion
	if i.enableReporting {
		res := &model.UpdateResult{}
		for deviceID := range deviceIDs {
			deviceIDStr := string(deviceID)
			res.MatchedCount += 1
			if deviceIDStr != "" {
				res.UpdatedCount += 1
				i.triggerReindex(ctx, deviceIDStr)
			}
		}
	}
```